exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/json/values/ScalarListener.java

```diff
@@ -76,4 +79,33 @@ protected void setArrayNull() {
   protected UserException typeConversionError(String jsonType) {
     return loader.typeConversionError(schema(), jsonType);
   }
+
+  /**
+   * Adds a field's most recent value to the pagination map.  This is necessary for the HTTP plugin
+   * for index or keyset pagination where the API transmits values in the results that are used to
+   * generate the next page.
+   *
+   * This data is only stored if the pagination map is defined, and has keys.
```

Comment:
Can this be rewritten in terms of generic column listeners rather than pagination and the HTTP plugin?

Reply:
Done!